addons/hr_timesheet/report/report_timesheet_templates.xml

```diff
@@ -7,14 +7,14 @@
                 <table class="table table-sm">
                     <thead>
                         <tr>
+                            <t t-set="timesheet_record_label" t-value="'Task' if show_task else 'Project' if show_project and not show_task else False"/>
```

Comment:
This will not be translatable. I suggest doing the other solution Xavier gave, but with the labels as text:
```suggestion
                            <t t-set="timesheet_record_label" t-value="False"/>
                            <t t-if="show_task">
                                <t t-set="timesheet_record_label">Task</t>
                            </t>
                            <t t-elif="show_project">
                                <t t-set="timesheet_record_label">Project</t>
                            </t>
```